### PR TITLE
Implements --repo flag to commands fetch, install, inspect, upgrade

### PR DESCRIPTION
**Pull request by sushilkm**

### CODE REVIEW

**Comment by thomastaylor312** [comment]
@sushilkm I have labeled this based on the current state of the PR. Let me know when you are ready to have this reviewed and I will add the appropriate label.

**Comment by sushilkm** [comment]
@thomastaylor312 I have completed the code, I would look for the feedback if it is the right way to do the stuff, I am still to add tests for the new code that I have added

**Comment by technosophos** [comment]
@sushilkm Should we remove `WIP` from the title and review it?

**Comment by thomastaylor312** [comment]
@technosophos is this going in to 2.4 or 2.5?

File: pkg/repo/chartrepo.go [hunk 1]
```diff
@@ -177,3 +177,52 @@ func (r *ChartRepository) generateIndex() error {
...
+// without adding repo to repostiories
+func FindChartInRepoURL(repoURL, chartName, chartVersion, certFile, keyFile, caFile string, getters []getter.Prop) (string, error) {
+
+	// Download and write the index file to a temporary location
+	tempIndexFile, err := ioutil.TempFile("", "tmp-repo-file")
+	if err != nil {
+		return "", fmt.Errorf("cannot write index file for repository requested")
+	}
+	defer func() {
+		os.Remove(tempIndexFile.Name())
```
Comment (fibonacci1729) [inline pkg/repo/chartrepo.go]: You can ```defer os.Remove(tempIndexFile.Name())``` here directly.

**Review by fibonacci1729 — APPROVED** [review]

**Pull request merged**